### PR TITLE
add arr.js and divs class sum, number, choose

### PR DESCRIPTION
**Pull request by veronaten**
Button Number and Button Choose don't work.

### CODE REVIEW

File: scripts/arr.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+
+let arr = [2, 5, 3, 9];
+let result = (arr[0]*arr[1]) + (arr[2]*arr[3]);
+alert(result);
```
Comment (magicsd) [inline scripts/arr.js]: It should be function since you expect it to be run. And again: use `let` only when needed. Use full naming for variables. Array should be given as an argument to the main function.

One function do the work (your task), another executes it and interacts with user.
```suggestion

function sumMultiply(array) {
  if (!Array.isArray(array) || array.length !== 4) {
    return 'Array is expected to have 4 numbers';
  }
  
  return array[0] * array[1] + array[2] * array[3];
}

function runSumMultiply() {
  const userArray = prompt('Enter array:', '[2, 5, 3, 9]') || [];
  
  const result = sumMultiply(userArray);
  
  alert(result);
}

// runSumMultiply - this is the function that should be run on button click

```

File: scripts/pow.js [hunk 2]
```diff
@@ -19,3 +19,9 @@ function runPow() {
       alert (`n должно быть целым и больше 1`);
   }
 }
+
+
+
+let arr = [2, 5, 3, 9];
+let result = (arr[0]*arr[1]) + (arr[2]*arr[3]);
+alert(result);
```
Comment (magicsd) [inline scripts/pow.js]: ???

File: style.css [hunk 3]
```diff
@@ -43,4 +43,27 @@ button:active {
...
+  margin-left: 20px;
+}
+
+.choose {
+  display: flex;
+  align-items: center;
+}
+
+.choose button {
+  margin-left: 20px;
```
Comment (magicsd) [inline style.css]: Why do you need it?

File: index.html [hunk 4]
```diff
@@ -15,6 +16,21 @@ <h2>Basic scripts</h2>
         <button onclick="runPow()" type="button">Run pow()</button>
       </div>
 
+      <h2>Array</h2>
+      <div class="sum">
+        <div>2*5+3*9=</div>
+        <button onclick="alert(result)" type="button">Result</button>
+      </div>
```
Comment (magicsd) [inline index.html]:
```suggestion
      <h4>Arrays</h4>
      
      <div class="item">
        <div>[0] * [1] + [2] * [3] =</div>
        <button onclick="runSumMultiply()" type="button">Run sumMultiply()</button>
      </div>
```

File: scripts/arr.js (hunk 1 above)
Comment (veronaten) [reply scripts/arr.js]: I don't understand this function sumMultiply(array).
And what does it mean:
Array.isArray(array) ?
array.length ?
Why we make  return 'Array is expected to have 4 numbers';  ?

File: scripts/pow.js (hunk 2 above)
Comment (veronaten) [reply scripts/pow.js]: I solved these tasks as shown in the examples in the book.. very easy)

File: style.css (hunk 3 above)
Comment (veronaten) [reply style.css]: I agree, this is unnecessary, I will fix it

File: index.html (hunk 4 above)
Comment (veronaten) [reply index.html]: Thank you.

File: scripts/pow.js (hunk 2 above)
Comment (magicsd) [reply scripts/pow.js]: What does this code do in this file?

File: scripts/arr.js (hunk 1 above)
Comment (magicsd) [reply scripts/arr.js]: https://frontendrockers.slack.com/archives/D016GU73WRM/p1610839367004200

File: scripts/arr.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+function dayNumber(week) {
+  const week = {1: 'Пн', 2: "Вт", 3: "Ср", 4: "Чт", 5: "Пт", 6: "Сб", 7: "Вскр"};
+  return week[userDay];
+}
+function runDayNumber() {
+  const userDay = Number(prompt('Введите число дня недели', [3]));
+  const show = dayNumber(userDay);
+  alert(show);
+}
```
Comment (magicsd) [inline scripts/arr.js]:
```suggestion
function sumMultiply(array) {
  if (!Array.isArray(array) || array.lenth !==4) {
    return 'Array is expected to have 4 numbers';
  }

  return array[0] * array[1] + array[2] * array [3];
}

function runSumMultiply() {
  const userArray = prompt('Enter array:', '[2, 5, 3, 9]') || [];

  const result = sumMultiply(userArray);

  alert(result);
} 

const week = { 1: 'Пн', 2: "Вт", 3: "Ср", 4: "Чт", 5: "Пт", 6: "Сб", 7: "Вскр" };

function runDayNumber() {
  const userDayNumber = Number(prompt('Введите число дня недели', 3) || 0);
  
  if (Number.isNaN(userDayNumber)) {
    alert('Enter valid number please')
    
    return
  }
  
  const dayName = week[userDayNumber];
  
  alert(`Here's your day: ${dayName}`);
}
```

File: scripts/arr.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
+function sumMultiply(array) {
+  if (!Array.isArray(array) || array.lenth !==4) {
```
Comment (magicsd) [inline scripts/arr.js]: ???
```suggestion
  if (!Array.isArray(array) || array.length !== 4) {
```

File: scripts/arr.js [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    return 'Array is expected to have 4 numbers';
+  }
+
+  return array[0] * array[1] + array[2] * array [3];
+}
+function runSumMultiply() {
```
Comment (magicsd) [inline scripts/arr.js]: ???
```suggestion
}

function runSumMultiply() {
```

File: index.html [hunk 8]
```diff
@@ -4,6 +4,7 @@
   <head>
     <link rel="stylesheet" href="style.css">
     <script src="scripts/pow.js"></script>
+    <script src="sctipts/arr.js"></script>
```
Comment (magicsd) [inline index.html]: 🤦 
```suggestion
    <script src="scripts/arr.js"></script>
```

File: scripts/arr.js [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+  const userArray = prompt('Enter array:', '[2, 5, 3, 9]') || [];
+
+  const result = sumMultiply(userArray);
```
Comment (magicsd) [inline scripts/arr.js]: Explain why this change is needed
```suggestion
  const userNumbers = prompt('Enter 4 numbers divided by space:', '2 5 3 9') || '0';
  
  const array = userNumbers.split(' ')

  const result = sumMultiply(array);
```

File: index.html (hunk 8 above)
Comment (veronaten) [reply index.html]: I'm burning with shame!! Sorryyy)

Comment (veronaten) [reply index.html]: Because I've already checked to see if I've set the path correctly.

File: scripts/arr.js (hunk 6 above)
Comment (veronaten) [reply scripts/arr.js]: I'm sorry.. this is my inattention..
I'm working on getting better)

File: scripts/arr.js (hunk 9 above)
Comment (veronaten) [reply scripts/arr.js]: So we adjusted the user input. Add the split method to split a string into substrings. When the user enters numbers with spaces, the code works
If the user enters the numbers in their own way, it will not be possible to calculate array. 
The code is working!!!

Comment (magicsd) [reply scripts/arr.js]: The reson is: `prompt` function returns a string, not an object. That's why we have to receive an input easy enough to transform into an array.

Comment (veronaten) [reply scripts/arr.js]: Yes)) Thank you!!

File: index.html [hunk 10]
```diff
@@ -4,6 +4,8 @@
   <head>
     <link rel="stylesheet" href="style.css">
     <script src="scripts/pow.js"></script>
+    <script src="scripts/arr.js"></script>
+    <script src="scripts/if.js"></script>
```
Comment (magicsd) [inline index.html]:
```suggestion

```

Comment (veronaten) [reply index.html]: Yes)

File: scripts/arr.js [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+  const userNumbers = prompt('Enter 4 numbers divided by space:', '2 5 3 9') || '0';
+
+  const array = userNumbers.split(' ');
+
+  const result = sumMultiply(array);
+
+  alert(result);
+} 
+
+const week = { 1: 'Пн', 2: 'Вт', 3: 'Ср', 4: 'Чт', 5: 'Пт', 6: 'Сб', 7: 'Вскр' }
```
Comment (magicsd) [inline scripts/arr.js]:
```suggestion

```

File: scripts/arr.js [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+
+    return
+  }
+
+  const dayName = week[userDayNumber];
+
+  alert(`Here's your day: ${dayName}`);
+}
+
+const weekDay = {1: 'Mn', 2: 'Ts', 3: 'Wd', 4: 'Th', 5: 'Fr', 6: 'St', 7: 'Sd' }
```
Comment (magicsd) [inline scripts/arr.js]:
```suggestion
const weekDaysByLanguage = {
  eng: ['Mn', 'Ts', 'Wd', 'Th', 'Fr', 'St', 'Sd'],
  ru: ['Пн', 'Вт', 'Ср', 'Чт', 'Пт', 'Сб', 'Вскр'],
};
```

File: scripts/arr.js [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+  const userLang = prompt('Select a language: "ru" or "eng"', 'ru');
+
+  if(userLang === 'ru') {
+    alert(`Here's your day: ${dayNumber}`);
+  } else if (userLang === 'eng') {
+    alert(`Here's your day: ${dayName}`);
+  } else {
+    alert('Select a language and enter "ru" or "eng"');
+  }
+}
```
Comment (magicsd) [inline scripts/arr.js]:
```suggestion
function revealTheDay() {
  const userLang = prompt('Select a language: "ru" or "eng"', 'ru');

  let dayNumberQuestionString = 'Enter the number of the day of the week';

  if (userLang === 'ru') {
    dayNumberQuestionString = 'Введите номер дня недели';
  }

  const userDayNumber = Number(prompt(dayNumberQuestionString, 3) || 0);

  const isDayNumberValid = !Number.isNaN(userDayNumber) && userDayNumber > 0 && userDayNumber <= 7

  if (!isDayNumberValid) {
    alert(userLang === 'ru' ? 'Введено неверное значение' : 'Enter valid number please');

    return;
  }

  const dayNames = weekDaysByLanguage[userLang];
  const userDayName = dayNames[userDayNumber - 1];
  // or userDayName = weekDaysByLanguage[userLang][userDayNumber - 1];

  const alertMessage = userLang === 'ru' ? 'Вот ваш день' : 'Here\'s your day';

  alert(`${alertMessage}: ${userDayName}`);
}
```

File: index.html [hunk 14]
```diff
@@ -15,6 +17,21 @@ <h2>Basic scripts</h2>
...
+      </div>
+
+      <div class="item">
+        <div>Day of the week:</div>
+        <button onclick="runDayNumber()" type="button">runDayNumber()</button>
+      </div>
+
+      <div class="item">
+        <div>Choose the day of week:</div>
+        <button onclick="dayChoose()" type="button">chooseLang()</button>
```
Comment (magicsd) [inline index.html]:
```suggestion
        <div>Reveal the day of week:</div>
        <button onclick="revealTheDay()" type="button">revealTheDay()</button>
```

File: scripts/arr.js (hunk 11 above)
Comment (veronaten) [reply scripts/arr.js]: Сan I not delete const week, as I use it in my the second task?

File: scripts/arr.js (hunk 13 above)
Comment (veronaten) [reply scripts/arr.js]: It's great! Thank you very much)

File: scripts/arr.js (hunk 11 above)
Comment (magicsd) [reply scripts/arr.js]: You can transform it into an array and use in both places.
```javascript
const rusWeekDayNames = ['Пн', 'Вт', 'Ср', 'Чт', 'Пт', 'Сб', 'Вскр']

// ...

const dayName = rusWeekDayNames[userDayNumber - 1];

// ...

const weekDaysByLanguage = {
  eng: ['Mn', 'Ts', 'Wd', 'Th', 'Fr', 'St', 'Sd'],
  ru: rusWeekDayNames,
};
```

File: scripts/arr.js [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+  'eng': ['Mn', 'Ts', 'Ws', 'Th', 'Fr', 'St', 'Sd'],
+  'ru': ['Пн', 'Вт', 'Ср', 'Чт', 'Пт', 'Сб', 'Вскр']
+};
+
+function revealTheDay() {
+  const userLang = prompt('Select a language: "ru" or "eng"', 'ru');
+
+  let dayNumberQuestionString = 'Enter the number the day of the week';
+
+  if(userLang === 'ru') {
```
Comment (magicsd) [inline scripts/arr.js]:
```suggestion
  if (userLang === 'ru') {
```

File: scripts/arr.js [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+    dayNumberQuestionString = 'Введите номер дня недели';
+  }
+
+  const userDayNumber = Number(prompt(dayNumberQuestionString, 3) || 0);
+
+  const isDayNumberValid = !Number.isNaN(userDayNumber) && userDayNumber > 0 && userDayNumber <= 7;
+
+  if(!isDayNumberValid) {
```
Comment (magicsd) [inline scripts/arr.js]:
```suggestion
  if (!isDayNumberValid) {
```

File: scripts/arr.js (hunk 11 above)
Comment (veronaten) [reply scripts/arr.js]: Precisely!

**Review by magicsd — APPROVED** [review]
